inc/abstractitiltarget.class.php

```diff
@@ -428,12 +428,23 @@ protected function setTargetUrgency($data, $formanswer) {
       }
       if (!is_null($urgency) && $urgency != 0) {
          $data['urgency'] = $urgency;
-         $data['priority'] = CommonITILObject::computePriority($data['urgency'], $data['impact']);;
       }
 
       return $data;
    }
 
+   /**
+    * Calculate priority from the predefined urgency and impact
+    *
+    * @param array $data
+    * @return array the updated data for the ITIL object
+    */
+   protected function setTargetPriority(array $data): array {
+      $data['priority'] = CommonITILObject::computePriority($data['urgency'], $data['impact']);
```

Comment:
You do not need to compute the priority, it will already be done in GLPI's core:

![image](https://user-images.githubusercontent.com/42734840/221892865-142d0415-0e2e-4b78-8fc8-60fabba69939.png)

All we need to do is to unset `$input['priority']` to force its computation:

```suggestion
      // Remove default priority so it can be computed
      if (isset($data['urgency']) || isset($data['impact'])) {
         unset($data['priority']);
      }
```

Reply:
More details: the current version fail because you are calling `CommonITILObject::computePriority` which will never work because of the following code:

![image](https://user-images.githubusercontent.com/42734840/221896212-928afd4b-52b2-4536-a90a-5f4a630cc10c.png)

`static::MATRIX_FIELD` will always be empty from `CommonITILObject`:

![image](https://user-images.githubusercontent.com/42734840/221896408-ded08972-5990-47ae-a039-78c9b83a1cfc.png)

So you would need to call it with this final item (`Ticket`, `Problem` or `Change`) or you wont be able to read the matrix.

With that being said, I think my original suggestion of letting the core handle it still the way to go.
If the process is already done in `prepareInputForAdd`, we don't need to duplicate it here.